CHANGELOG.md

```diff
@@ -6,9 +6,156 @@
+a different set of selectors was used in the Notebook 6 and older.
+Users with the custom CSS file using the selectors from legacy Notebook versions
+will need to update them and note that tutorials may refer to the selectors for the old Notebook versions.
+```
+
+### Diagrams in Markdown
+
+Matching GitHub-Flavoured Markdown, JupyterLab 4.1 now supports [Mermaid](https://github.com/mermaid-js/mermaid) diagrams.
+To create a mermaid diagram use the `mermaid` language specifier for a code block in a markdown cell or document, for example:
+
+~~~
+```mermaid
+flowchart LR
+
+A[Hard] -->|Text| B(Round)
+B --> C{Decision}
+C -->|One| D[Result 1]
+C -->|Two| E[Result 2]
+```
+~~~
+
+which renders as:
+
+<img alt="Rendered Mermaid diagram going from left to right with modern look; the diagram contains blueish square node with text Hard pointing to round square node with text Round; over the arrow connecting the two nodes there is a word Text; the second node further connects to a rhombus-shaped node with text Decision which then connects to two further nodes, Result 1 and Result 2." src="https://raw.githubusercontent.com/krassowski/jupyterlab/changelog-4.1/docs/source/getting_started/changelog_assets/4.1-merimad-diagram.png" class="jp-screenshot">
+
+
+### Inline Completer
+
+JupyterLab now supports completion presented as ghost text in the cell and file editors,
+enabling multi-line completions such as those generated by large language models (like GPT-based copilots).
+The suggestions are provided by plugins implementing the ``IInlineCompletionProvider`` API;
+by default a single provider which uses kernel history is available.
+
+<img alt="In a code cell with `def fac` content a ghost text containing a suggestion representing further code of factorial function is shown; over the code cell there is a floating widget allowing to accept the suggestion and iterate between alternative suggestions" src="https://raw.githubusercontent.com/krassowski/jupyterlab/changelog-4.1/docs/source/getting_started/changelog_assets/4.1-inline-completer.png" class="jp-screenshot">
+
+The suggestions can be invoked as-you-type or manually using a configurable shortcut (by default <kbd>Alt</kbd> + <kbd>\\</kbd>).
+The default keyboard shortcuts are displayed in the small widget shown when hovering over the ghost suggestion:
+- <kbd>Alt</kbd> + <kbd>End</kbd> - accept suggestion
+- <kbd>Alt</kbd> + <kbd>[</kbd> - previous suggestion
+- <kbd>Alt</kbd> + <kbd>]</kbd> - next suggestion
+
+To enable the inline suggestions based on the kernel history, go to Settings → Inline Completer → History provider → check the "enabled" checkbox.
+
+The Inline Completer API is still considered experimental and may be subject to changes, please share feedback!
+
+### Keyboard navigation improvements
+
+Numerous improvements to keyboard navigation with focus on accessibility and usability are included in this release:
+- the notebook cells now retain focus
```

Comment:
Under what circumstances do notebook cells now retain focus when they didn't before?

Reply:
Almost every, especially when switching to command mode and all actions associated with it. Previously focus always reverted to notebook nice. We can probably just link to related blog post rather than explain in detail here as the audience for whom it is relevant will understand from these brief notes

Reply:
I added link to the blogpost below, it explains it all in much more detail.